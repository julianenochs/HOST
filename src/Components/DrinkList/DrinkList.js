import React, { Component } from 'react';
import './DrinkList.scss';
import { connect } from 'react-redux';
import { togglePartyInput, saveCocktailToParty, deleteCocktail } from '../../Actions/index';

export class DrinkList extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      selectedParty: props.parties.list.id,
      showSubmission: false
    }
  }

  handleDelete = (id) => {
    this.props.deleteCocktail(id)
  }

  handleChange = (e) => {
    e.stopPropagation();
    this.setState({[e.target.name]: e.target.value});
  }

  handleSubmit = () => {
    this.props.saveCocktailsToParty({selectedParty: this.state.selectedParty, cocktail: this.props.selectedCocktail});
    this.setState({ showSubmission: true })
    setTimeout(() => {
      this.setState({ showSubmission: false })
    }, 2000)
  }

  render() {
    const { viewCocktail, parties, fromParty } = this.props
    return(
      <section className='single-cocktail__container'>
        <div className='cocktail-card'>
          <h2 className='cocktail-card-name'>{viewCocktail.strDrink}</h2>
          <img className='cocktail-card__img' src={`${viewCocktail.strDrinkThumb}`} alt={`${viewCocktail.strDrink}`}/>
          <h3>Glass Type: {viewCocktail.strGlass}</h3>
          <p>How To: {viewCocktail.strInstructions}</p>
          {fromParty ?
            <button onClick={() => this.handleDelete(viewCocktail.idDrink)}> Remove </button>
          :
            parties.list.length ?
          <div>
            <h3>
              Save To Party:
            </h3>
              <select name='selectedParty' onChange={this.handleChange} value={this.state.selectedParty}>
                <option value='select'> Select </option>
              {parties.list.map(party => {
                return <option value={`${party.id}`}>{`${party.name}`}</option>
              })}
            </select>
            <button className='' onClick={this.handleSubmit}> Submit </button>
            {this.state.showSubmission ? <div id='submitted' > Submitted </div> : ''}
          </div>
            : <button className='add-party__button' onClick={this.props.togglePartyInput}> Start a Party </button>}
        </div>
      </section>
    )
  }
}

const mapStateToProps = state => ({
  parties: state.parties,
  isPartyInputVisible: state.isPartyInputVisible,
  selectedCocktail: state.selectedCocktail
})

const mapDispatchToProps = dispatch => ({
  togglePartyInput: () => dispatch(togglePartyInput()),
  saveCocktailsToParty: (cocktail) => dispatch(saveCocktailToParty(cocktail)),
  deleteCocktail: (cocktailId) => dispatch(deleteCocktail(cocktailId))
});

export default connect(mapStateToProps, mapDispatchToProps)(DrinkList);
