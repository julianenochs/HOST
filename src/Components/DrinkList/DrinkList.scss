.single-cocktail__container {
  align-items: center;
  display: flex;
  height: 85vh;
  justify-content: center;
  width: 85%;
}

.cocktail-card {
  align-items: center;
  background: rgb(92,92,92);
  background: linear-gradient(0deg, rgba(92,92,92,1) 0%, rgba(92,92,92,1) 35%, rgba(148,148,148,0.7301295518207283) 100%);display: flex;
  border-radius: 5px;
  flex-direction: column;
  font-family: 'Federo', sans-serif;
  height: fit-content;
  letter-spacing: 2px;
  padding: 15px;
  width: 35%;
}

.cocktail-card__img {
  height: 200px;
  width: 200px;
}