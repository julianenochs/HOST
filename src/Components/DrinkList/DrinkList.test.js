import React from 'react';
import { DrinkList } from './DrinkList';
import { shallow } from 'enzyme';

describe('Drink List', () => {
  let wrapper;
  let mockViewCocktail = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<DrinkList 
      props={'hi'}
      key={3}
      viewCocktail={mockViewCocktail}
      />)
  });

  it.skip('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});