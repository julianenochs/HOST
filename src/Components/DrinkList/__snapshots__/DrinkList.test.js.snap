// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drink List should match the snapshot 1`] = `undefined`;
